sh/jshn.sh

```diff
@@ -291,11 +303,11 @@ json_select() {
 	local type
 	local cur
 
-	[ -z "$1" ] && {
+	[ -z "$target" ] && {
```

Comment:
Those helper variables containing the function parameters are used sometimes but not always. Now it is a design question if they should be used everywhere, nowhere are only in some functions for some reason.

Reply:
Again, trying to make things consistent.

Reply:
In most cases, they are declared because they could be rewritten within the function. Another way to make it consistent would be to replace it by a comment in the function header and by the parameter index here because it is not reassigned in this particular function.